### DISCUSSION TITLE
Totum V5

### DISCUSSION THREAD

**Discussion by ttm-alexeypolunin** [totumonline/totum-ru-issues-and-discussions#597] [Анонсы]
В MIT и PRO репозитории выложена V5. Инструкции по переходу здесь: https://ru.docs.totum.online/update_four_five

Ключевое изменение — теперь Totum PRO платный (начиная с 11 пользователя), а MIT-версия урезана по функционалу.

**Из MIT убраны:**

- Файлы (поля типа Файл)
- Tree-view
- Поворот таблицы
- Анонимные таблицы
- Формы
- Карточки и Канбан
- JSON-API (Remotes продолжат работать, но если вы использовали Remotes через JSON-API то он тоже будет отключен)
- Подключение внешних функций
- execSSH
- Сервисы генерации .docx/.xlsx

Вобщем MIT остается для обучения и пет-проектов.

**В PRO добавлен:**

- Модуль на GO, обрабатывающий данные пользователей и обеспечивающий мониторинг таблицы для каждого подключения (он же смотрит за лицензиями).
- Исправлены 100500 ошибок.
- Он же будет использоваться для некоторого дальнейшего функционала (например непрерывных Кронов, Кэшей внешних подключений, Батчинга входящих запросов, Очередей отправки внешних запросов итп)

Стоимость на лицензии на странице PRO: https://ru.totum.online/pro

Если система до 10 пользователей, то лицензия не нужна. 

По умолчанию 1 пользователь = 1 одновременное подключение (один браузер). Это настраивается, но каждое дополнительное потребляет лицензионный слот.

Некоторый функционал доступен только с платной лицензией на любое количество пользователей:

- Интеграция с Only Office
- Daemons
- Сервисы

Описание работы с лицензиями в документации: https://ru.docs.totum.online/pro-install#pro-license

Купить лицензию: https://license.totum-online.ru/Forms/register (в РФ оплата только от юрлиц)

### Если вы переключаете рабочую систему

0. Заранее купите лицензию
1. Делайте это в рабочее время
2. Перед тем как приступать, удостоверьтесь, что мы на связи и сможем вам помочь, написав в Телеграм на @alexeypolunin

**Comment by LostAlly** [comment]
Цены вроде вполне адекватные.
А можно поподробней расписать про внешних пользователей, в прайсе написано 5 таблиц, но непонятно как это будет выглядеть на практике. Если бесплатная лицензия до 10 пользователей, ограниченных нужно покупать? Если мне например нужно 10 штук таких будет, пока мне до запуска в прод еще далековато но хотелось бы делать с учетом таких ограниченных пользователей.

**Reply by ttm-alexeypolunin** [reply]
Ограниченный пользователь (limit user) это такой же пользователь как и прочие.

Тотум выделяет их только для целей лицензирования. Если у пользователя 8 или менее таблиц в доступе (есть 3 технические: Лог, Нотификации, Загрузка xlsx), то модуль лицензии может относить этого пользователя к упрощенной лицензии, в случае, если не хватает полноценных лицензионных слотов.

Те ограниченные пользователи существуют только для целей экономии денег на стоимости лицензии.

В каких случаях: когда есть условно ключевые сотрудники - это эмпирически 30-40 таблиц в доступе. И есть большая масса сотрудников, контракторов, клиентов с ограниченным доступом (таблица доставок/заказов, таблица отгрузок/приемки).

В этом случае, спроектировав решение таким образом, что бы у них было менее 8 таблиц (не обязательно одинаковых) вы можете сэкономить на лицензиях.

Например: 40 сотрудников офиса и кабинеты 2000 точек выдачи.

или 30 диспетчеров и 2000 водителей/курьеров.

В приведенных случаях вы можете поступить двумя путями: 

1. Купив безлимит за 400 тр/год и надо платить за год (что может напрягать тк большую часть пользователей в этих случаях составляют аккаунты с наименьшим участием в системе)
2. Купив 60 полных лицензий и 2000 лимитных 316 тр/год и можно платить по 3 месяца.
3. Можно спроектировать систему не предоставляя авторизационного доступа контрагентам внутрь. Например если нужно показывать статус заказа клиенту по ссылке, то можно обойтись анонимными таблицами. Они потребляют 1 лицензионный слот - пользователя anonym. Или формами - тоже могут потреблять один слот.

**Comment by LostAlly** [comment]
Еще вопросики по поводу лицензирования.

1. На станице с ценами есть кнопка регистраци, нужно зарегистрироваться, чтобы пользоваться бесплатной версией.

2. Планируется такая схема разработки работы, на одном впс схема для разработки и отдельная схема для тестирования куда будут накладываться обновления и после тестирования будет изменяться рабочий сервер, после обновления рабочего, он вместе с боевыми данными будет копироваться в тестовую. И отдельный выделеный впс для рабочей базы. Можно это все делать под одной лицензией или тут какие то сложные схемы? Пока у нас три сотрудника и все попадает под бесплатную лицензию. Возможно когда запустим рабочий сервер их уже будет больше, но не больше 10 пока это точно.

может сделаете чат в группе телеги? С входом только для зарегистрированных пользователей, чтобы можно было быстро общаться между собой без расхода времени на модерацию.

**Reply by ttm-alexeypolunin** [reply]
До 10 пользователей включительно файл лицензии не проверяется за исключением некоторого функционала. Сейчас это подключение к Only Office, из ближайшего это daemons (непрерывные кроны) и наверное большая часть функционала, которая будет выходить дальше будет требовать обязательную действующую лицензию.

В одной лицензии доступно два хоста, они могут быть на разных серверах.

**Reply by ttm-alexeypolunin** [reply]
Чат в телеге мы делать не будем тк это требует затрат на модерацию этого чата, иначе там очень быстро начнут толкать колеса и шмаль. Более того, как только мы выпустим бот, отвечающий на вопросы по документации - команда разработки перестанет отвечать здесь на технические вопросы аккаунтам, не имеющим ни одной действующей лицензии.

**Comment by OAndrey1985** [comment]
При попытке обновления на этапе создания GOM -service
ln -s /home/totum/totum-mit/totum-gom.service /etc/systemd/system/totum-gom.service && systemctl daemon-reload && systemctl enable totum-gom && service totum-gom start && service totum-gom status
failed to create symbolic link '...totum-gom.service' permision denided

**Reply by ttm-alexeypolunin** [reply]
это надо выполнять от root

**Reply by OAndrey1985** [reply]
спасибо, разобрался
